src/storage/src/sink/kafka.rs

```diff
@@ -1578,16 +1578,11 @@ fn evaluate_partition_by(partition_by: &MirScalarExpr, row: &[Datum]) -> u64 {
     // the best available option. The behavior is clearly documented in the
     // user-facing `CREATE SINK` docs.
     let temp_storage = RowArena::new();
-    match partition_by.eval(row, &temp_storage) {
-        Ok(hash) => match hash {
-            Datum::Int32(i) => i.try_into().unwrap_or(0),
-            Datum::Int64(i) => i.try_into().unwrap_or(0),
-            Datum::UInt32(u) => u64::from(u),
-            Datum::UInt64(u) => u,
-            _ => unreachable!(),
-        },
```

Comment:
Are there any Kafka sinks where the partition-by expression evaluates to one of the non-uint64 types in any catalog? Then this would be a regression vs. the current state as we'd partition to 0.

Reply:
It would be very surprising to me if so, I looked at the original PR for this feature and there has always been an explicit cast to UInt64 https://github.com/MaterializeInc/materialize/pull/33618/files#diff-c840a7fbb6a642d6cbf71dddd92ec1b6dfa86310022ead5673e8fa509ba73dc4R3871-R3874

Reply:
good idea